lib/unparser/emitter/literal/primitive.rb

```diff
@@ -33,12 +33,21 @@ class Complex < self
 
           RATIONAL_FORMAT = 'i'.freeze
 
-          MAP = IceNine.deep_freeze(
-            Float    => :float,
-            Rational => :rational,
-            Fixnum   => :int,
-            Bignum   => :int
-          )
+          MAP =
+            if 0.class == Integer
```

Comment:
I'd prefer we use `#equal?` over `#==` here. As identity is the least powerful primitive.